src/tools/validation.rs

```diff
@@ -20,7 +20,11 @@ where
 
     static_assertions::const_assert_eq!(HEADER_VERSION, 5);
     let header = read_index_header(path)?;
-    let blob_size = path.metadata()?.len();
+    let blob_path = path.with_extension("blob");
+    if !blob_path.exists() {
+        return Err(Error::index_header_validation_error("blob file doesn't exist").into());
```

Comment:
This should not be an error. This should be a warning and the validation should be started anyway and obviously should not fail on invalid blob size (for simplicity, you can pass size from the index header if blob is not exists)